File: b15904.cpp
#include <iostream>
#include <string>

int main()
{
	std::string sentences;
	const char ucpc[4]{ 'U', 'C', 'P', 'C' }
	std::getline(std::cin, sentences);
	short start_idx = 0;
	bool flag = false;
	for (auto& sentence : sentences)
	{
		if (sentence == ucpc[start_idx]) {
			start_idx++;
		}
		if (start_idx == 4) {
			flag = true;
			break;
		}
	}

	if (flag) {
		std::cout << "I love UCPC" << std::endl;
	}
	else {
		std::cout << "I hate UCPC" << std::endl;
	}
	return 0;
}

File: greedy/python/b15904.py
sentences = input()
result = []
UCPC = "UCPC"
start_idx = 0
flag = False
for ch in sentences :
    if ch == UCPC[start_idx] :
        result.append(ch)
        start_idx += 1
    if start_idx == 4 :
        flag = True
        break

if flag :
    print("I love UCPC")
else : 
    print("I hate UCPC")